content/en/agent/faq/linux-agent-2022-key-rotation.md

```diff
@@ -0,0 +1,124 @@
+---
+
+As a common best practice, Datadog periodically rotates the keys and certificates used to sign Datadog's Agent packages. The following GPG keys, used to sign the Agent RPM and DEB packages, reach their end-of-life in June 2022 and will be rotated in April 2022:
+
+- The RPM signing key with hash [`A4C0B90D7443CF6E4E8AA341F1068E14E09422B3`][1] will be rotated in April 4th 2022 and will be replaced by the key with hash [`C6559B690CA882F023BDF3F63F4D1729FD4BF915`][2]
+- The DEB signing key with hash [`A2923DFF56EDA6E76E55E492D3A80E30382E94DE`][3] will be rotated in April 18th 2022 and will be replaced by the key with hash `D75CEA17048B9ACBF186794B32637D44F14F620E`[4]
+
+Customers using Datadog's RPM or DEB packages might require a manual action to import the new key on their systems in order to install or upgrade the Agent after the rotation takes place.
+
+<div class="alert alert-info">
+**Note**: This DOES NOT affect the functionality of already running Agents, and only limits the ability to install or upgrade to a newer version of the Agent. This also doesn't affect Windows or MacOS Agents, nor Dockerized Linux Agents.
+</div>
+
+If you're using one of the following install methods, your hosts trust the key automatically and no further action is needed:
+
+* [Agent install script][5] v1.5.0 or later (released Jun 29, 2021)
+* [Chef cookbook][6] v4.11.0 or later (released Aug 10, 2021)
+* [Ansible role][7] v4.9.0 or later (released May 6, 2021)
+* [Puppet module][8] v3.13 or later (released Aug 11, 2021)
+* [SaltStack formula][9] v3.4 or later (released Aug 12, 2021)
+* [Elastic Beanstalk][10] config templates updated as of Mar 29, 2021 (should contain `DATADOG_RPM_KEY_FD4BF915.public` under `gpgkey`)
+* Containerized Agents (Docker/Kubernetes): No action needed on any version
+* Heroku: No action needed on any version.
+* Windows/MacOS Agents: No action needed on any version.
+
+Additionally, if you are using the DEB version of the Agent version 7.31.0 or greater, your hosts should have installed the `datadog-signing-keys` package that automatically trusts the new key.
+
+If unsure, read the steps in [Check if a host trusts the new GPG key](#check-if-a-host-trusts-the-new-gpg-key).
+
+Trying to install or upgrade the Agent package without trusting the new key results in `NOKEY` errors when installing the RPM package and `NO_PUBKEY` errors when installing the DEB package.
+
+For hosts running older versions of the install methods listed above or older versions of the DEB package, Datadog recommends updating the install method to the latest version. Alternatively, Debian and Ubuntu users can update the Agent to version 7.31.0 or greater. Otherwise, the keys can be manually updated on host by following these instructions:
+
+### Manual update
+
+{{< tabs >}}
+{{% tab "Debian/Ubuntu" %}}
+
+Run the following commands on the host:
+
+```bash
+$ wget -qO - https://keys.datadoghq.com/DATADOG_APT_KEY_F14F620E.public | sudo apt-key add -
```

Comment:
So while this would work, technically it's not ideal because the usage of `apt-key` for 3rd part repos is discouraged per https://wiki.debian.org/DebianRepository/UseThirdParty#OpenPGP_Key_distribution

I understand that we're doing this to accomodate for people who haven't yet adopted the approach of using `signed-by` option for the Datadog DEB repository, so I guess it's ok to keep this, but we should at least provide a clear note about this not being a good way to go. Alternatively, we could provide link to the in-app installation instructions at https://app.datadoghq.com/account/settings#agent/debian which do the whole thing with keyring and `signed-by`.

Reply:
The problem with the better way is that the `datadog.list` contents change depending on the Agent major version, and that would make the docs quite more complex... futhermore, if you have use a config management tool, it will rewrite the file on the next run.

And now that I think about it, this is a problem for RPM as well :/

Reply:
Hmm, that's a good point. I think this should be highlighted in the docs - if you're using an older version of a cfg management tool, your hand changes will be overwritten.

In terms of RPM platforms, importing the key to the `rpm` database will work for package-level signatures. For repodata signatures, it will work on SUSE, because zypper respects rpm-imported GPG keys when validating repodata signatures. However, for `yum`/`dnf` this is not true - people who modify repofile manually on a RH platform, but then rerun a cfg management tool and clear yum/dnf cache will not have the new key for repodata verification.

Reply:
On the other hand, repodata validation is only turned on by install methods that already add the `CURRENT` key to the list of keys, so we don't need to worry about that. I think we can just `rpm --import` (but we would need to test all distros and different versions of config management tools...).

Reply:
Oh, that's right! Well, at least that makes it easier for us :)